sgl-kernel/CMakeLists.txt

```diff
@@ -0,0 +1,290 @@
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -O3 -DNDEBUG \
+    -DOPERATOR_NAMESPACE=${OPERATOR_NAMESPACE} \
+    -Xcompiler -fPIC \
+    -std=c++17 \
+    -DFLASHINFER_ENABLE_F16 \
+    -DCUTLASS_ENABLE_TENSOR_CORE_MMA=1 \
+    -DCUTLASS_VERSIONS_GENERATED \
+    -DCUTE_USE_PACKED_TUPLE=1 \
+    -DCUTLASS_TEST_LEVEL=0 \
+    -DCUTLASS_TEST_ENABLE_CACHED_RESULTS=1 \
+    -DCUTLASS_DEBUG_TRACE_LEVEL=0 \
+    --ptxas-options=-v \
+    --expt-relaxed-constexpr \
+    -Xcompiler=-Wconversion \
+    -Xcompiler=-fno-strict-aliasing"
+)
+
+# Add architecture-specific flags.
+foreach(ARCH ${CMAKE_CUDA_ARCHITECTURES})
+    set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_${ARCH},code=sm_${ARCH}")
+endforeach()
+
+# Check for SM90A support.
+if((CUDA_VERSION_MAJOR GREATER_EQUAL 12 AND CUDA_VERSION_MINOR GREATER_EQUAL 0) OR DEFINED ENV{ENABLE_SM90A})
+    if("$ENV{ENABLE_SM90A}" STREQUAL "ON" OR "$ENV{ENABLE_SM90A}" STREQUAL "1")
+        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_90a,code=sm_90a")
+        add_compile_definitions(ENABLE_SM90A)
+    endif()
+endif()
+
+# Check for SM100A support.
+if((CUDA_VERSION_MAJOR GREATER_EQUAL 12 AND CUDA_VERSION_MINOR GREATER_EQUAL 8) OR DEFINED ENV{ENABLE_SM100A})
+    if("$ENV{ENABLE_SM100A}" STREQUAL "ON" OR "$ENV{ENABLE_SM100A}" STREQUAL "1")
+        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_100,code=sm_100")
+        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_100a,code=sm_100a")
+    else()
+        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -use_fast_math")
+    endif()
+endif()
+
+# Check for FP8 support.
+if(DEFINED ENV{ENABLE_FP8})
+    if("$ENV{ENABLE_FP8}" STREQUAL "ON" OR "$ENV{ENABLE_FP8}" STREQUAL "1")
+        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} \
+            -DFLASHINFER_ENABLE_FP8 \
+            -DFLASHINFER_ENABLE_FP8_E4M3 \
+            -DFLASHINFER_ENABLE_FP8_E5M2"
+        )
+        add_compile_definitions(ENABLE_FP8)
+    endif()
+endif()
+
+# Check for BF16 support.
+if(DEFINED ENV{ENABLE_BF16})
+    if("$ENV{ENABLE_BF16}" STREQUAL "ON" OR "$ENV{ENABLE_BF16}" STREQUAL "1")
+        set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -DFLASHINFER_ENABLE_BF16")
+        add_compile_definitions(ENABLE_BF16)
+    endif()
+endif()
+
+# Remove problematic CUDA flags.
+string(REPLACE "-D__CUDA_NO_HALF_OPERATORS__" "" CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS}")
+string(REPLACE "-D__CUDA_NO_HALF_CONVERSIONS__" "" CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS}")
+string(REPLACE "-D__CUDA_NO_BFLOAT16_CONVERSIONS__" "" CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS}")
+string(REPLACE "-D__CUDA_NO_HALF2_OPERATORS__" "" CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS}")
+
+# Create a more complete pytorch_extension_utils.h file
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/pytorch_extension_utils.h" "#pragma once
+#include <torch/extension.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <sstream>
+#include <vector>
+
+// Common check macros
```

Comment:
Why include these common check macros in CMake?

Reply:
ok, removed.